include/ignition/math/PiecewiseScalarField3.hh

```diff
@@ -0,0 +1,215 @@
+        {
+          if (pieces[i].region.Empty())
+          {
+            std::cerr << "Region #" << i << " (" << pieces[i].region
+                      << ") in piecewise scalar field definition is empty."
+                      << std::endl;
+          }
+          for (size_t j = i + 1; j < pieces.size(); ++j)
+          {
+            if (pieces[i].region.Intersects(pieces[j].region))
+            {
+              std::cerr << "Detected overlap between regions in "
+                        << "piecewise scalar field definition: "
+                        << "region #" << i << " (" << pieces[i].region
+                        << ") overlaps with region #" << j << " ("
+                        << pieces[j].region << "). Region #" << i
+                        << " will take precedence when overlapping."
+                        << std::endl;
+            }
+          }
+        }
+      }
+
+      /// \brief Define piecewise scalar field as `_field` throughout R^3 space
+      public: static PiecewiseScalarField3 Throughout(ScalarField3T _field)
+      {
+        return PiecewiseScalarField3<ScalarField3T, ScalarT>({
+            {Region3<ScalarT>::Unbounded, std::move(_field)}});
+      }
+
+      /// \brief Evaluate the piecewise scalar field at `_p`
+      /// \param[in] _p piecewise scalar field argument
+      /// \return the result of evaluating `F(_p)`, or NaN
+      ///   if the scalar field is not defined at `_p`
+      public: ScalarT Evaluate(const Vector3<ScalarT> &_p) const
+      {
+        auto it = std::find_if(
+            this->pieces.begin(), this->pieces.end(),
+            [&](const Piece &piece) {
+              return piece.region.Contains(_p);
+            });
+        if (it == this->pieces.end())
+        {
+          return std::numeric_limits<ScalarT>::quiet_NaN();
+        }
+        return it->field(_p);
+      }
+
+      /// \brief Call operator overload
+      /// \see PiecewiseScalarField3::Evaluate()
+      /// \param[in] _p piecewise scalar field argument
+      /// \return the result of evaluating `F(_p)`, or NaN
+      ///   if the scalar field is not defined at `_p`
+      public: ScalarT operator()(const Vector3<ScalarT> &_p) const
+      {
+        return this->Evaluate(_p);
+      }
+
+      /// \brief Compute the piecewise scalar field minimum
+      /// Note that, since this method computes the minimum
+      /// for each region independently, it implicitly assumes
+      /// continuity in the boundaries between regions, if any.
+      /// \param[out] _pMin scalar field argument that yields
+      ///   the minimum, or NaN if the scalar field is not
+      ///   defined anywhere (i.e. default constructed)
+      /// \return the scalar field minimum, or NaN if the
+      ///   scalar field is not defined anywhere (i.e. default
+      ///   constructed)
+      public: ScalarT Minimum(Vector3<ScalarT> &_pMin) const
+      {
+        if (this->pieces.empty()) {
```

Comment:
```suggestion
        if (this->pieces.empty())
        {
```

Reply:
Done in 38cbc349121a83d2b94df6328f581fa38aeb127e.